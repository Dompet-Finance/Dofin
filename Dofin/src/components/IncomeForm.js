import React from 'react';
import {
  View,
  TextInput,
  Button,
  Text
} from 'react-native';

class IncomeForm extends React.Component {

  pressTest() {
    return true
  }

  render() {
    return (
      <View>
        <Text>Income</Text>
        <TextInput />
        <TextInput />
        <Button title="" onPress={() => this.pressTest()}/>
      </View>
    )
  }
};

export default IncomeForm;
