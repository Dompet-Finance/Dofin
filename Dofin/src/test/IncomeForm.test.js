import React from 'react';
import { shallow } from 'enzyme';
import {
  View,
  TextInput,
  Button
} from 'react-native';

import IncomeForm from '../components/IncomeForm';

describe('<IncomeForm />', () => {
  const wrapper = shallow(<IncomeForm />);

  it('should render a Text', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should be a view component', () => {

    expect(wrapper.type()).toEqual(View);
  });

  it('should render TextInput', () => {

    expect(wrapper.find(TextInput)).toHaveLength(2)
  });

  it('should render Button', () => {

    expect(wrapper.find(Button)).toHaveLength(1)
  });

  test('it should simulate click save on button', () => {
    const button = wrapper.find(Button);
    const prop = button.prop('onPress');
    expect(prop).toBeInstanceOf(Function);
    expect(prop()).toEqual(true)
  })
});
