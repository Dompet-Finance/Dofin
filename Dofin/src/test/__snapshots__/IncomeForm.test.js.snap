// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncomeForm /> should render a Text 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
    >
        Income
    </Text>
    <TextInput />
    <TextInput />
    <Button
        onPress={[Function]}
        title=""
    />
</View>,
  "nodes": Array [
    <View>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
      >
            Income
      </Text>
      <TextInput />
      <TextInput />
      <Button
            onPress={[Function]}
            title=""
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <IncomeForm />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": IncomeForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
          >
                    Income
          </Text>
          <TextInput />
          <TextInput />
          <Button
                    onPress={[Function]}
                    title=""
          />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
          >
                    Income
          </Text>
          <TextInput />
          <TextInput />
          <Button
                    onPress={[Function]}
                    title=""
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <IncomeForm />,
}
`;
